docker/openemr/5.0.3/autoconfig.sh

```diff
@@ -51,6 +51,30 @@ auto_setup() {
     fi
 }
 
+#Turn on API from docker
+if [ "$ACTIVATE_API" == "yes" ]; then
+
+    USER="openemr"
+    PASSWORD="openemr"
+    HOST="mysql"
+    DATABASE="openemr"
+    
+    if [ "$MYSQL_USER" != "" ]; then
+        USER=$MYSQL_USER
+    fi
+    if [ "$MYSQL_PASS" != "" ]; then
+        PASSWORD=$MYSQL_PASS
+    fi
+    if [ "$MYSQL_HOST" != "" ]; then
+        HOST=$MYSQL_HOST
+    fi
+    if [ "$MYSQL_DATABASE" != "" ]; then
+        DATABASE=$MYSQL_DATABASE
+    fi
+        
+    mysql -u $USER  --password=$PASSWORD -h "$HOST" -e "UPDATE globals SET gl_value = 1 WHERE gl_name = \"rest_api\"" $DATABASE
+fi    
```

Comment:
Lets place this and integrate this into the auto_setup() a bit more (also make variables more specific), so that function will look something like (this approach also allows easy addition of other settings that require database changes):
```
auto_setup() {

    CONFIGURATION="server=${MYSQL_HOST} rootpass=${MYSQL_ROOT_PASS} loginhost=%"
    if [ "$MYSQL_ROOT_USER" != "" ]; then
        CONFIGURATION="${CONFIGURATION} root=${MYSQL_ROOT_USER}"
    fi
    if [ "$MYSQL_USER" != "" ]; then
        CONFIGURATION="${CONFIGURATION} login=${MYSQL_USER}"
        CUSTOM_USER="$MYSQL_USER"
    else
        CUSTOM_USER="openemr"
    fi
    if [ "$MYSQL_PASS" != "" ]; then
        CONFIGURATION="${CONFIGURATION} pass=${MYSQL_PASS}"
        CUSTOM_PASSWORD="$MYSQL_PASS"
    else
        CUSTOM_PASSWORD="openemr"
    fi
    if [ "$MYSQL_DATABASE" != "" ]; then
        CONFIGURATION="${CONFIGURATION} dbname=${MYSQL_DATABASE}"
        CUSTOM_DATABASE="$MYSQL_DATABASE"
    then
        CUSTOM_DATABASE="openemr"
    fi
    if [ "$OE_USER" != "" ]; then
        CONFIGURATION="${CONFIGURATION} iuser=${OE_USER}"
    fi
    if [ "$OE_PASS" != "" ]; then
        CONFIGURATION="${CONFIGURATION} iuserpass=${OE_PASS}"
    fi

    chmod -R 600 .
    php auto_configure.php -f ${CONFIGURATION} || return 1

    echo "OpenEMR configured."
    CONFIG=$(php -r "require_once('/var/www/localhost/htdocs/openemr/sites/default/sqlconf.php'); echo \$config;")
    if [ "$CONFIG" == "0" ]; then
        echo "Error in auto-config. Configuration failed."
        exit 2
    fi

    if [ "$ACTIVATE_API" == "yes" ]; then
        mysql -u "$CUSTOM_USER"  --password="$CUSTOM_PASSWORD" -h "$MYSQL_HOST" -e "UPDATE globals SET gl_value = 1 WHERE gl_name = \"rest_api\"" "$CUSTOM_DATABASE"
    fi
}
```

Reply:
Hello @bradymiller , I think I have fixed this